java/fury-core/src/test/java/io/fury/serializer/ExternalizableSerializerTest.java

```diff
@@ -18,49 +18,17 @@
 
 import static org.testng.Assert.assertEquals;
 
-import com.google.common.base.Preconditions;
 import io.fury.Fury;
 import io.fury.config.Language;
+import io.fury.serializer.test.Factory;
 import java.io.Externalizable;
-import java.io.IOException;
-import java.io.ObjectInput;
-import java.io.ObjectOutput;
-import lombok.Data;
 import org.testng.annotations.Test;
 
 public class ExternalizableSerializerTest {
 
-  @Data
-  public static class A implements Externalizable {
-    private int x;
-    private int y;
-    private byte[] bytes;
-
-    @Override
-    public void writeExternal(ObjectOutput out) throws IOException {
-      out.writeInt(x);
-      out.writeInt(y);
-      out.writeInt(bytes.length);
-      out.write(bytes);
-    }
-
-    @Override
-    public void readExternal(ObjectInput in) throws IOException, ClassNotFoundException {
-      this.x = in.readInt();
-      this.y = in.readInt();
-      int len = in.readInt();
-      byte[] arr = new byte[len];
-      Preconditions.checkArgument(in.read(arr) == len);
-      this.bytes = arr;
-    }
-  }
-
   @Test
   public void testExternalizable() {
-    A a = new A();
-    a.x = 1;
-    a.y = 1;
-    a.bytes = "bytes".getBytes();
+    Externalizable a = Factory.newA(1, 1, "bytes".getBytes());
```

Comment:
Is there some reason for using `Factory` to create `A` instead of invoking constructor directly?

Reply:
In my code the issue was only reproducible when `A` was in a different package and had `package` visibility. So from the package of the test the constructor would not be callable.

Reply:
I see, thanks. How about adding a factory method in `A`, just move `newA` to `A`

Reply:
The class `A` would still not be public and thus not visible to the test.

Please note that in my case I think the problem is that my serialized objects contain objects where the class implements `Externalizable` and there also the `writeReplace()` and `readResolve()` methods. This may explain why the class is not public.

Reply:
I see

Reply:
Could we add some comments on `A` and `Factory` to indicate why we need the factory and a inassessible class?

Reply:
Certainly.